common/codex32.c

```diff
@@ -13,6 +13,186 @@
 #include <secp256k1_schnorrsig.h>
 #include <time.h>
 
+const uint8_t LOG[32] =
+{
+     0,  0,  1, 14,  2, 28, 15, 22,
+     3,  5, 29, 26, 16,  7, 23, 11,
+     4, 25,  6, 10, 30, 13, 27, 21,
+    17, 18,  8, 19, 24,  9, 12, 20,
+};
+
+const uint8_t LOG_INV[31] =
+{
+     1,  2,  4,  8, 16,  9, 18, 13,
+    26, 29, 19, 15, 30, 21,  3,  6,
+    12, 24, 25, 27, 31, 23,  7, 14,
+    28, 17, 11, 22,  5, 10, 20,
+};
+
+static void addition_gf32(uint8_t *x, uint8_t y)
+{
+	*x = *x ^ y;
+	return;
+}
+
+static void multiply_gf32(uint8_t *x, uint8_t y)
+{
+	if (*x == 0 || y == 0) {
+		*x = 0;
+	} else {
+		*x = LOG_INV[(LOG[*x] + LOG[y]) % 31];
+	}
+	return;
+}
+
+/* Helper to input a single field element in the checksum engine. */
+static void input_fe(struct checksum_engine *checksum_engine, uint8_t e)
```

Comment:
I would get rid of struct checksum_engine, and  hand the (const u8 *generator, u8 *residue, size_t table_size).